crates/forge-runner/src/lib.rs

```diff
@@ -0,0 +1,594 @@
+        fuzzer_runs: u32,
+        fuzzer_seed: u64,
+    ) -> Self {
+        Self {
+            workspace_root,
+            exit_first,
+            fork_targets,
+            fuzzer_runs,
+            fuzzer_seed,
+        }
+    }
+}
+
+pub struct RunnerParams {
+    corelib_path: Utf8PathBuf,
+    contracts: HashMap<String, StarknetContractArtifacts>,
+    predeployed_contracts: Utf8PathBuf,
+    environment_variables: HashMap<String, String>,
+    linked_libraries: Vec<LinkedLibrary>,
+}
+
+impl RunnerParams {
+    #[must_use]
+    pub fn linked_libraries(&self) -> &Vec<LinkedLibrary> {
+        &self.linked_libraries
+    }
+    #[must_use]
+    pub fn corelib_path(&self) -> &Utf8PathBuf {
+        &self.corelib_path
+    }
+}
+
+impl RunnerParams {
+    #[must_use]
+    pub fn new(
+        corelib_path: Utf8PathBuf,
+        contracts: HashMap<String, StarknetContractArtifacts>,
+        predeployed_contracts: Utf8PathBuf,
+        environment_variables: HashMap<String, String>,
+        linked_libraries: Vec<LinkedLibrary>,
+    ) -> Self {
+        Self {
+            corelib_path,
+            contracts,
+            predeployed_contracts,
+            environment_variables,
+            linked_libraries,
+        }
+    }
+}
+
+pub struct CancellationTokens {
+    exit_first: CancellationToken,
+    error: CancellationToken,
+}
+
+impl CancellationTokens {
+    #[must_use]
+    pub fn new() -> Self {
+        let exit_first = CancellationToken::new();
+        let error = CancellationToken::new();
+        Self { exit_first, error }
+    }
+}
+
+impl Default for CancellationTokens {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// Exit status of the runner
+#[derive(Debug, PartialEq, Clone)]
+pub enum RunnerStatus {
+    /// Runner exited without problems
+    Default,
+    /// Some test failed
+    TestFailed,
+    /// Runner did not run, e.g. when test cases got skipped
+    DidNotRun,
+}
+
+#[derive(Debug, Clone)]
+pub struct ForkConfig {
+    url: Url,
+    block_id: BlockId,
+}
+
+impl ForkConfig {
+    #[must_use]
+    pub fn new(url: Url, block_id: BlockId) -> Self {
+        Self { url, block_id }
+    }
+}
+
+impl TryFrom<RawForkParams> for ForkConfig {
+    type Error = anyhow::Error;
+
+    fn try_from(value: RawForkParams) -> std::result::Result<Self, Self::Error> {
+        Ok(ForkConfig {
+            url: value.url.parse()?,
+            block_id: value.block_id,
+        })
+    }
+}
+
+impl ForkConfigTrait for ForkConfig {}
+
+pub type TestCase = CollectedTestCase<ForkConfig>;
+
+#[derive(Debug, Clone)]
+pub struct TestCrate {
+    sierra_program: Program,
+    test_cases: Vec<TestCase>,
+}
+
+impl TestCrate {
+    #[must_use]
+    pub fn new(sierra_program: Program, test_cases: Vec<TestCase>) -> Self {
+        Self {
+            sierra_program,
+            test_cases,
+        }
+    }
+}
+
+pub trait TestCaseFilter {
+    fn should_be_run(&self, test_case: &TestCase) -> bool;
+}
```

Comment:
Everything from the beginning of the file to this line should be moved to separate files

Reply:
In a separate PR